File: Data Structures & Algorithms/palindromic-substrings/submission-1.go
func countSubstrings(s string) int {
    var expand func(left int, right int) int
    expand = func(left int, right int) int {
        count := 0

        for left >= 0 && right < len(s) && s[left] == s[right] {
            count++
            left--
            right++
        }
        return count
    }
    res := 0
    for i := range s {
        // Odd palindrome
        res += expand(i, i)
        // Even palindrome
        res += expand(i, i+1)
    }
    return res
}

File: Data Structures & Algorithms/spiral-matrix/submission-1.py
class Solution:
    def spiralOrder(self, matrix: List[List[int]]) -> List[int]:
        rows = len(matrix)
        cols = len(matrix[0])

        res = []

        top, bottom = 0, rows - 1
        left, right = 0, cols - 1

        while top <= bottom and left <= right:
            # top: left -> right
            for col in range(left, right+1):
                res.append(matrix[top][col])
            top += 1

            # right: top -> bottom
            for row in range(top, bottom+1):
                res.append(matrix[row][right])
            right -= 1

            if top <= bottom:
                # bottom: right -> left
                for col in range(right, left-1, -1):
                    res.append(matrix[bottom][col])
                bottom -= 1
            if left <= right:
                # left: bottom -> top
                for row in range(bottom, top-1, -1):
                    res.append(matrix[row][left])
                left += 1
        return res